src/reducers/reducerUtil.js

```diff
@@ -44,6 +44,19 @@ function handlePagedAction(action, state) {
                 cursor: cursor
             });
         }
+        case ActionTypes.SINGLE: {
+            let _contents = state.items;
+            _contents = {
+                ..._contents,
```

Comment:
this *copies*  the existing contents and then sets the result to an additional key based on the payload data. 

from my experience this won't create duplicates.